feat(exit): add system register read and write exit reasons

luodeb opened the pull request:
- Add SysRegRead and SysRegWrite variants to the AxVCpuExitReason enum
- SysRegRead contains address and register number for system register reads
- SysRegWrite contains address, register number, and data for system register writes

luodeb committed:
feat(exit): add system register read and write exit reasons

- Add SysRegRead and SysRegWrite variants to the AxVCpuExitReason enum
- SysRegRead contains address and register number for system register reads
- SysRegWrite contains address, register number, and data for system register writes

aarkegz commented on src/exit.rs:
```diff
@@ -95,6 +95,20 @@ pub enum AxVCpuExitReason {
     },
+    SysRegRead {
+        /// The address of the system register read.
+        addr: usize,
```
The name `addr` is kind of confusing. It's not a memory addr.

luodeb replied:
ok

aarkegz commented on src/exit.rs:
```diff
@@ -95,6 +95,20 @@ pub enum AxVCpuExitReason {
+        /// The address of the system register read.
+        addr: usize,
+        /// The system register number.
+        reg: usize,
```
This is **NOT** the number of the system register, according to your code in `arm_vcpu`, it's the number of the GPR.

luodeb replied:
ok

aarkegz commented on src/exit.rs:
```diff
@@ -95,6 +95,20 @@ pub enum AxVCpuExitReason {
+        /// The address of the system register write.
+        addr: usize,
+        /// The system register number.
+        reg: usize,
```
Do we need both `reg` and `value` here?

luodeb replied:
It seems like it's indeed a repetition.

luodeb committed:
docs(exit): improve documentation for system register access

- Clarify the usage of 'addr' field for different architectures
- Update comments to provide more precise information

aarkegz reviewed (approved):
LGTM
